akka-http/src/main/scala/akka/http/scaladsl/unmarshalling/sse/LineParser.scala

```diff
@@ -51,14 +51,20 @@ private final class LineParser(maxLineSize: Int) extends GraphStage[FlowShape[By
               // Lookahead for LF after CR
               case `cr` if at < bs.length - 1 && bs(at + 1) == lf ⇒
                 parseLines(bs, at + 2, at + 2, parsedLines :+ bs.slice(from, at).utf8String)
+              // if starts with a CR, but we don't know the next byte, we can't slice it yet, so simply advance
```

Comment:
We can ;) But we also need to return a flag to remember that the last character read was a `CR`.